### ISSUE TITLE
Failed to load config "react-app" to extend from.

### ISSUE THREAD

**Issue by bartlomiejzuber** [eslint/eslint#13283]
Hey, after upgrade to v7.0.0 I see those build errors:
![image](https://user-images.githubusercontent.com/9931428/81562436-20ec7900-9395-11ea-9671-0e06931d00ab.png)

**Tell us about your environment**

* **ESLint Version: v7.0.0**
* **Node Version: v13.14.0**
* **npm Version: v13.14.0**
* **react-scripts Version: v3.4.1**

**What parser (default, Babel-ESLint, etc.) are you using?**
The one react-scripts are using, `default` I guess.

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
  "extends": ["react-app"],
  "plugins": [
    "react-hooks"
  ],
  "rules": {
    "react-hooks/rules-of-hooks": "error",
    "react-hooks/exhaustive-deps": "warn",
    "no-use-before-define": [
      "error",
      {
        "functions": true,
        "classes": true,
        "variables": true
      }
    ]
  }
}

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

It's thrown from all components insde src.

```bash
  yarn eslint "src/**"
```

**What did you expect to happen?**
Should do the checks as v. 6 did.

**What actually happened? Please include the actual, raw output from ESLint.**
```
./src/components/StandardDialogContent/StandardDialogContent.tsx
Failed to load config "react-app" to extend from.
Referenced from: /home/runner/work/emagine/emagine/.eslintrc
```

**Are you willing to submit a pull request to fix this bug?**
Yeah, although I don't know if it's a eslint bug or intended behavior and instead CRA is the better place to fix this issue.

**Comment by kaicataldo** [comment]
> Should do the checks as v. 7 did.

To be clear, do you mean as ESLint v6 behaved?

My guess is that you might need to use the new `--resolve-plugins-relative-to` CLI flag since [plugin resolution has changed in v7](https://eslint.org/docs/user-guide/migrating-to-7.0.0#plugin-resolution-has-been-updated).

**Comment by bartlomiejzuber** [comment]
>To be clear, do you mean as ESLint v6 behaved?

Ahh sorry, yes I meant v6. :)

Thanks for sharing. I found out that the root cause is probably the change of default ignore pattern from `/node_modules/*` to `/**/node_modules/*`. I've checked the location of `eslint-config-react-app` and it's under `node_modules\react-scripts\node_modules` which means it being ignored.

Problem can be solved via:
```
yarn add eslint-config-react-app @typescript-eslint/eslint-plugin@2.x @typescript-eslint/parser@2.x babel-eslint@10.x eslint@6.x eslint-plugin-flowtype@4.x eslint-plugin-import@2.x eslint-plugin-jsx-a11y@6.x eslint-plugin-react@7.x eslint-plugin-react-hooks@2.x
```
But I think it's far from ideal.

Due to the fact that I am running eslint through `react-scripts` I think I can't pass `--resolve-plugins-relative-to` to the eslint. Couldn't find any option to set this inside `.eslintrc` as well. Currently the only way I see it to eject the `CRA`(which I'd like not to do). If you have some other ideas, please share. I will try to find out some cleaner solution over the weekend.

**Comment by mysticatea** [comment]
> I found out that the root cause is probably the change of default ignore pattern

Ignore patterns affect only lint targets, so it's not related to config/plugin loading.

---

Would you provide concrete repro steps?
I'd like to debug it, but I'm not sure how to reproduce it.

I tried:

```
npx create-react-app foo
cd foo
npm i eslint@latest
npx eslint "src/*"
```

but it works well (it should use `"extends": "react-app"` setting).

**Comment by ldrick** [comment]
I found the problem, I hope. And I confirm @bartlomiejzuber seeing `eslint-config-react-app` being under `node_modules\react-scripts\node_modules`.

This happens only if you use yarn for setup. npm brings `eslint-config-react-app` directly under node_modules.

I did:

> npx create-react-app appy --template typescript
> cd app
> npm i eslint@latest
> removed folder node_modules
> remove yarn.lock
> removed package.lock
> removed eslint entry in package.json
> added in package.json new script "lint": "eslint \"src/**\""
> added .eslintrc.js
```javascript
module.exports = {
  root: true,
  extends: ['react-app'],
};
```
> yarn
> yarn lint

**Comment by bartlomiejzuber** [comment]
> Ignore patterns affect only lint targets, so it's not related to config/plugin loading.

Yes, that's wrong path.

>This happens only if you use yarn for setup. npm brings eslint-config-react-app directly under node_modules.
That's right, just tried and indeed problem exists when using `yarn` instead of `npm`. Nice catch @ldrick . 👍

**Comment by mysticatea** [comment]
```
npx create-react-app appy --template typescript
cd app
npm i eslint@latest
removed folder node_modules
remove yarn.lock
removed package.lock
removed eslint config entry in package.json
added in package.json new script "lint": "eslint "src/**""
added .eslintrc.js
```

That makes the error is the correct behavior.
Because the `app/package.json` doesn't have `eslint-config-react-app`, it means `require("eslint-config-react-app")` go to fail, but your `.eslintrc.js` depends on `eslint-config-react-app`.

In fact, the popular package managers have "dedupe" behavior, and the side-effect of the "dedupe" sometimes changed `require(foo)` for indirect dependencies, as a result, `require("eslint-config-react-app")` looks working accidentally.

In your case, installing the new eslint changed the "dedupe"'s side-effect, and you got the error. You need `yarn add eslint-config-react-app` and some peer deps to use it.

**Comment by ldrick** [comment]
Thanks @mysticatea , so maybe it is better to ask in create react app project to add this to their docs https://create-react-app.dev/docs/setting-up-your-editor/#experimental-extending-the-eslint-config
